app/assets/javascripts/data_sets/editTable.js.coffee

```diff
@@ -30,26 +34,110 @@ setupTable = (cols, data) ->
     forceFitColumns: true
     rowHeight: 35
     syncColumnCellResize: true
+
   for x, i in data
     data[i]['id'] = i
 
   view = new Slick.Data.DataView()
   grid = new Slick.Grid '#slickgrid-container', view, cols, options
+  currId = data.length
+
+  view.setItems data
 
   grid.onClick.subscribe (e, args) ->
     cell = grid.getCellFromEvent e
     if cell.cell == grid.getColumns().length - 1
+      grid.getEditorLock().commitCurrentEdit()
+      grid.resetActiveCell()
       item = view.getItem cell.row
       view.deleteItem item.id
       grid.invalidate()
+
+      if grid.getDataLength() == 0
+        fields = {id: currId}
+        currId += 1
+        for x in cols
+          fields[x.field] = ''
+        view.addItem fields
+
   view.onRowCountChanged.subscribe (e, args) ->
     grid.updateRowCount()
     grid.render()
+
   view.onRowsChanged.subscribe (e, args) ->
     grid.invalidateRow args.rows
     grid.render()
 
-  view.setItems data
+  $('.edit_table_add').click ->
+    if $('#edit_table_save_1').hasClass 'disabled'
+      return
+
+    grid.getEditorLock().commitCurrentEdit()
+    grid.resetActiveCell()
+
+    fields = {id: currId}
+    currId += 1
+    for x in cols
+      fields[x.field] = ''
+    view.addItem fields
+    grid.scrollRowIntoView view.getLength()
+
+  $('.edit_table_save').click ->
+
+    # commit the current edit regardless of whether or not we can actually upload
+    grid.getEditorLock().commitCurrentEdit()
+    grid.resetActiveCell()
+
+    # check if we've already started saving
+    if $('#edit_table_save_1').hasClass 'disabled'
```

Comment:
I'm wondering what the difference between the IDs edit_table_save_1 and edit_table_save is? Maybe there's a better way to distinguish between them